src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -96,7 +101,7 @@ public PriorityQueue(int initialCapacity, IComparer<TPriority>? comparer)
                 ? Array.Empty<(TElement, TPriority)>()
                 : new (TElement, TPriority)[initialCapacity];
 
-            Comparer = comparer ?? Comparer<TPriority>.Default;
+            _comparer = comparer;
```

Comment:
At this point I think we should also change some of the other constructors:
```C#
public PriorityQueue()
{
    _nodes = Array.Empty<(TElement, TPriority)>();
}

public PriorityQueue(IComparer<TPriority>? comparer) : this()
{
    _comparer = comparer;
}
```

Reply:
Actually, having read further on, we should probably leave them the way they are and change this line to be:
```C#
_comparer = comparer ?? typeof(TPriority).IsValueType ? null : Comparer<TPriority>.Default;
```
or something like that.  Given current optimizations in the JIT, it's better for reference types to just use _comparer and it's better for value types to use `Comparer<TPriority>.Default`.

Reply:
Or even:
```C#
if (typeof(TPriority).IsValueType)
{
    if (comparer != Comparer<TPriority>.Default)
    {
        _comparer = comparer;
    }
}
else
{
    _comparer = comparer ?? Comparer<TPriority>.Default;
}
```
and you could add a TODO with the relevant issue about improving this later if the JIT improves in this area.

Reply:
Nice, addressed.